sdk/trace/span.go

```diff
@@ -598,16 +599,15 @@ type samplingData struct {
 	remoteParent bool
 	parent       trace.SpanContext
 	name         string
-	cfg          *Config
+	sampler      Sampler
 	span         *span
 	attributes   []attribute.KeyValue
 	links        []trace.Link
 	kind         trace.SpanKind
 }
 
 func makeSamplingDecision(data samplingData) SamplingResult {
-	sampler := data.cfg.DefaultSampler
-	return sampler.ShouldSample(SamplingParameters{
+	return data.sampler.ShouldSample(SamplingParameters{
```

Comment:
Remove makeSamplingDecision and duplicate samplingData struct.

This has become a convoluted function and struct that duplicates `SamplingParameters` and embeds the singular method the function calls within the struct. It does not appear necessary anymore.

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/1706

Reply:
It'd be better to separate PR. Removing `makeSamplingDecision` will result in removing struct `samplingData` because it has no use.

Reply:
> It'd be better to separate PR.

Sorry, that wasn't clear :facepalm:, that was what I was hoping as well.
 
> Removing `makeSamplingDecision` will result in removing struct `samplingData` because it has no use.

Agreed, that would be ideal :+1: 

To be clear we should leave this for a follow on PR that resolves #1706